### DISCUSSION TITLE
Seperate generated do not edit files from file intended for manual edits

### DISCUSSION THREAD

**Discussion by wireva** [ent/ent#3376] [Ideas]
I think the current layout of the ent folder is quite messy. If you have a lot of models you will have a ton of crud files beside the few file you can actually edit

I don't know if this is technically possible because go packages seem to be limited to be one folder, but it would be way easier to navigate and keep track of the project if the non generated files would be seperated

A structure somewhat like this is what I imagine:
```
.
├── ent
│   ├── entc.go
│   ├── generated
│   │   ├── cat_create.go
│   │   ├── cat_update.go
│   │   ├── person_create.go
│   │   └── person_delete.go
│   └── schema
│       ├── cat.go
│       └── person.go
└── main.go

```

It would also help people who are new to the framework so they don't get overwhelmed by all the files in the same folder that they never need.

**Comment by a8m** [comment]
Hey @niwla23 👋 
The default location of the schema is under `ent/schema`, but you can change both it and `entc.go` location and set up a custom directory for the codegen target. See the `--target` option in the doc: https://entgo.io/docs/code-gen#code-generation-options and [this example](https://github.com/ent/ent/blob/master/entc/integration/gremlin/ent/generate.go#L7).

**Reply by wireva** [reply]
Thanks! Since I am using entc.go I have done it like this now:
```go
err := entc.Generate("./schema", &gen.Config{Target: "./generated", Package: "github.com/niwla23/lagersystem/manager/ent/generated"}, opts...)
```
